python/ql/src/experimental/dataflow/internal/DataFlowPublic.qll

```diff
@@ -148,6 +148,19 @@ class ParameterNode extends EssaNode {
   override DataFlowCallable getEnclosingCallable() { this.isParameterOf(result, _) }
 }
 
+/**
+ * A node that controls whether other nodes are evaluated.
+ */
+class GuardNode extends ControlFlowNode {
+  ConditionBlock conditionBlock;
+
+  GuardNode() { this = conditionBlock.getLastNode() }
+
+  predicate controlsNode(ControlFlowNode node, boolean testIsTrue) {
+    conditionBlock.controls(node.getBasicBlock(), testIsTrue)
```

Comment:
This looks big.

Reply:
Indeed, might it help to meld the definition of `GuardNode` into the definition of `BarrierGuard`?

Reply:
This predicate could just use `BasicBlock` as its interface rather than `ControlFlowNode`.

Reply:
Ah, that is an easy fix, thanks.